web/modules/custom/solution/src/EventSubscriber/WorkflowTransitionEventSubscriber.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace Drupal\solution\EventSubscriber;
+
+use Drupal\message\Entity\Message;
+use Drupal\message\Entity\MessageTemplate;
+use Drupal\message_notify\MessageNotifier;
+use Drupal\og\Entity\OgMembership;
+use Drupal\state_machine\Event\StateChangeEvent;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Event subscribers for Organic Groups.
+ */
+class WorkflowTransitionEventSubscriber implements EventSubscriberInterface {
```

Comment:
I guess we'll have many other events to respond and perform actions. From an architectural point of view wouldn't it be better to consider a consistent way to respond to events instead of just coding subscribers? I'm thinking to Rules module. It would be more elegant and consistent if we can manage all reactions there. The current approach is correct, but I'm thinking that it would be better to have a single point where we can  react.